pkg/solana/chainreader/chain_reader.go

```diff
@@ -217,54 +227,99 @@ func (s *SolanaChainReaderService) CreateContractType(readIdentifier string, for
 		return nil, fmt.Errorf("%w: no contract for read identifier", types.ErrInvalidConfig)
 	}
 
-	return s.bindings.CreateType(values.contract, values.readName, forEncoding)
+	return s.bindings.CreateType(values.contract, values.genericName, forEncoding)
 }
 
-func (s *SolanaChainReaderService) init(namespaces map[string]config.ChainReaderMethods) error {
-	for namespace, methods := range namespaces {
-		for methodName, method := range methods.Methods {
-			var idl codec.IDL
-			if err := json.Unmarshal([]byte(method.AnchorIDL), &idl); err != nil {
-				return err
-			}
-
-			idlCodec, err := codec.NewIDLAccountCodec(idl, config.BuilderForEncoding(method.Encoding))
-			if err != nil {
-				return err
-			}
-
-			s.lookup.addReadNameForContract(namespace, methodName)
+func (s *SolanaChainReaderService) addCodecDef(forEncoding bool, namespace, genericName string, readType codec.ChainConfigType, idl codec.IDL, idlDefinition interface{}, modCfg commoncodec.ModifiersConfig) error {
+	mod, err := modCfg.ToModifier(codec.DecoderHooks...)
```

Comment:
The ChainWriter should have similar logic to this, correct? Instead of having a codec for each program and adding modifiers to that codec, you have a codec entry for each method essentially

Reply:
Yes, same as the evm impl.